### PR TITLE
docs(framework) Update Flower tutorial series to Flower Next (part 4 of 4)

### PR DESCRIPTION
**Pull request by chongshenng**
### Description

Part 4 of the Flower tutorial series uses the `start_simulation()` approach, which is pre-Flower 1.8.

### Related issues/PRs

Related to #3295, #3313, #3314. 

### Explanation

This PR updates Part 4 of the Flower tutorial series.  Changes:
* Replace `start_simulation()` with `run_simulation()`. 
* Add the `ClientApp` and `ServerApp` objects, with accompanying descriptions.
* Update `client_fn` signature to use with `context`.
* Use `server_fn`. 

### Checklist

- [X] Implement proposed change
- ~~[ ] Write tests~~
- [X] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [X] Update the changelog entry below
- [X] Make CI checks pass

### Changelog entry

<examples>

### CODE REVIEW

File: doc/source/tutorial-series-customize-the-client-pytorch.ipynb [hunk 1]
```diff
@@ -294,7 +344,7 @@
    "source": [
     "This works as expected, two clients are training for three rounds of federated learning.\n",
     "\n",
-    "Let's dive a little bit deeper and discuss how Flower executes this simulation. Whenever a client is selected to do some work, `start_simulation` calls the function `numpyclient_fn` to create an instance of our `FlowerNumPyClient` (along with loading the model and the data).\n",
+    "Let's dive a little bit deeper and discuss how Flower executes this simulation. Whenever a client is selected to do some work, `run_simulation` launches the `ClientApp` object which in turn calls the function `numpyclient_fn` to create an instance of our `FlowerNumPyClient` (along with loading the model and the data).\n",
     "\n",
     "But here's the perhaps surprising part: Flower doesn't actually use the `FlowerNumPyClient` object directly. Instead, it wraps the object to makes it look like a subclass of `flwr.client.Client`, not `flwr.client.NumPyClient`. In fact, the Flower core framework doesn't know how to handle `NumPyClient`'s, it only knows how to handle `Client`'s. `NumPyClient` is just a convenience abstraction built on top of `Client`. \n",
```
Comment (danielnugraha) [inline doc/source/tutorial-series-customize-the-client-pytorch.ipynb:349]: Should we add a couple of sentences to the new Flower Next here?

**Review by danielnugraha — COMMENTED** [review]
Looks good, I only have two small comments.

File: doc/source/tutorial-series-customize-the-client-pytorch.ipynb [hunk 2]
```diff
@@ -40,7 +42,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "!pip install -q flwr[simulation] torch torchvision scipy"
+    "!pip install -q flwr-nightly[simulation] flwr-datasets[vision] torch torchvision scipy"
```
[the remark this answers is not in the document]
Comment (chongshenng) [reply doc/source/tutorial-series-customize-the-client-pytorch.ipynb:45]: I'll remove nightly after 1.9 release.

File: doc/source/tutorial-series-customize-the-client-pytorch.ipynb (hunk 1 above)
Comment (chongshenng) [reply doc/source/tutorial-series-customize-the-client-pytorch.ipynb:349]: Good point, I've added links to the guides on how to use `ClientApp` and the explainer doc for Flower Next.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**